pkg/cvo/metrics.go

```diff
@@ -156,21 +212,50 @@ func RunMetrics(runContext context.Context, shutdownContext context.Context, lis
 		} else {
 			select {
 			case <-runContext.Done(): // clean shutdown
-			case result := <-resultChannel: // crashed before a shutdown was requested
+			case result := <-resultChannel: // crashed before a shutdown was requested or metrics server recreated
+				if restartServer {
+					klog.Info("Creating metrics server with updated TLS configutation.")
+					server = createHttpServer()
+					go startListening(server, tlsConfig, listenAddress, resultChannel)
+					restartServer = false
+					continue
+				}
 				resultChannelCount--
 				if result.error == nil {
 					klog.Infof("Collected metrics %s goroutine.", result.name)
 				} else {
 					klog.Errorf("Collected metrics %s goroutine: %v", result.name, result.error)
 					loopError = result.error
```

Comment:
This probably needs to be updated to something like:

```go
if loopError != nil {
  klog.Error(loopError)
}
loopError = result.error
```

so that we don't silently forget about any `loopError` issues we're clobbering, like an earlier `Failed to create TLS config: ...`.

Reply:
I created a new method `handleServerResult` which resets `lastLoopError` if necessary. It also checks for and treats as a non-error `ErrServerClosed`, since this is always returned upon successful shutdown, as to not clobber our real shutdown reason which may be stored in `lastLoopError`. 
In the case of a `result.error != ErrServerClosed`, `lastLopError` is set to this error however the previous `lastLoopError` has already been logged so it is not clobbered. I considered constructing and returning a new `lastLoopError` from both and returning it but that seemed like overkill.

Reply:
`handleServerResult` handling in 0e6136c727b0 looks good to me.